tinygrad/runtime/ops_clang.py

```diff
@@ -10,13 +11,21 @@
 }[platform.system()]
 CLANG_PROGRAM_HEADER = '#include <math.h>\n#define max(x,y) ((x>y)?x:y)\n#define int64 long\n#define half __fp16\n#define uchar unsigned char\n#define bool uchar\n'
 class ClangProgram:
-  def __init__(self, name:str, prg:str):
-    prg = CLANG_PROGRAM_HEADER + prg
+  def __init__(self, name:str, prg:str, binary:bool=False):
     # TODO: is there a way to not write this to disk?
     fn = f"{tempfile.gettempdir()}/clang_{hashlib.md5(prg.encode('utf-8')).hexdigest()}.{args['ext']}"
-    if not os.path.exists(fn):
-      subprocess.check_output(args=('clang -shared -O2 -Wall -Werror -x c '+args['cflags']+' - -o '+fn+'.tmp').split(), input=prg.encode('utf-8'))
-      os.rename(fn+'.tmp', fn)
+    if not binary:
+      prg = CLANG_PROGRAM_HEADER + prg
+      # TODO: is there a way to not write this to disk?
+      if not os.path.exists(fn):
+        subprocess.check_output(args=('clang -shared -O2 -Wall -Werror -x c '+args['cflags']+' - -o '+fn+'.tmp').split(), input=prg.encode('utf-8'))
+        os.rename(fn+'.tmp', fn)
+    else:
+      if DEBUG >= 5: print(prg)
+      if getenv('ARM64'):
```

Comment:
Do you need this if, or is just the `else` on binary enough?

Reply:
Isn't it the case that currently binary is never set to true for the github action, so this else branch is never run for CI. Might be a case that I am missing something though